### PR TITLE
gcc-linaro toolchains: bump to 8.3-2019.03 and rename to gcc-arm

### PR DESCRIPTION
**Pull request by kszaq**
Align aarch64 toolchain with the version used to compile userspace.

Motivation: with recent kernels the toolchain version is one of the config options. If a different toolchain version is used, it sometimes causes having to manually select kernel options during build - this should not happen if the build is expected complete without user interaction.

Build- and runtime-tested with 5.1.0-rc1 kernel on Amlogic S905X, test with 3.14 pending.

I would like to request Allwinner/Rockchip tests for targets running aarch64 kernel and arm userspace - no hardware here.

### CODE REVIEW

**Comment by Kwiboo** [comment]
Please also update `gcc-linaro-arm-linux-gnueabihf` to the same version to keep them in sync.
Package names maybe should be changed from linaro to arm now that arm has taken over the linaro toolchain?

I will run build tests on Rockchip over night.

**Comment by kszaq** [comment]
@Kwiboo `gcc-linaro-arm-linux-gnueabihf` also updated.

**Comment by kszaq** [comment]
Runtime tested with Amlogic 3.14 kernel - works fine. I will amend variable and package names later today.

**Comment by kszaq** [comment]
PR and toolchains renamed. I also made an amendment to `config/path` to rename `KERNEL_LINARO_TOOLCHAIN` to `KERNEL_TOOLCHAIN`.

File: packages/linux/package.mk [hunk 1]
```diff
@@ -45,9 +45,9 @@ esac
 
 PKG_KERNEL_CFG_FILE=$(kernel_config_path)
 
-if [ -n "$KERNEL_LINARO_TOOLCHAIN" ]; then
-  PKG_DEPENDS_HOST="$PKG_DEPENDS_HOST gcc-linaro-$KERNEL_LINARO_TOOLCHAIN:host"
-  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET gcc-linaro-$KERNEL_LINARO_TOOLCHAIN:host"
+if [ -n "$KERNEL_TOOLCHAIN" ]; then
+  PKG_DEPENDS_HOST="$PKG_DEPENDS_HOST gcc-linaro-$KERNEL_TOOLCHAIN:host"
+  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET gcc-linaro-$KERNEL_TOOLCHAIN:host"
```
Comment (MilhouseVH) [inline packages/linux/package.mk]: s/linaro/arm/ ?

Comment (kszaq) [reply packages/linux/package.mk]: Changed, thanks for noticing!

**Comment by kszaq** [comment]
Verified to build and run fine after renaming patches with Amlogic mainline.

@Kwiboo Do you possibly have results for Rockchip?

**Pull request merged**